app/javascript/mastodon/reducers/compose.js

```diff
@@ -223,8 +225,7 @@ export default function compose(state = initialState, action) {
       .set('is_composing', false);
   case COMPOSE_SENSITIVITY_CHANGE:
     return state.withMutations(map => {
-      const isSpoilerActive = enableAlwaysShowSpoiler ? state.get('spoiler_text').length > 0 : state.get('spoiler');
-      if (!isSpoilerActive) {
+      if (!isSpoilerActive(state)) {
```

Comment:
I don't like that this stops you turning on sensitivity if the spoiler is active. That should never be an option but it's still peculiar logic. Why not split this in two:
```suggestion
      map.set('sensitive', !state.get('sensitive'));
      if (isSpoilerActive(state)) {
          map.set('sensitive', true);
      }
```
This way the logic is made explicit: this action flips the sensitivity. Except if the spoiler is active, tough luck, stays sensitive.